PeriodogramAnalysis/test_power_density_statistics.py

```diff
@@ -0,0 +1,96 @@
+from typing import Generator, Any
+
+import numpy as np
+import pytest
+from miv.core.datatype import Signal
+from power_density_statistics import (
+    SpectrumAnalysisBase,
+    SpectrumAnalysisPeriodogram,
+    SpectrumAnalysisWelch,
+)
+
+
+def mock_signal_generator():
+    timestamps = np.linspace(0, 10, 1000, endpoint=False)
+    signal = (
+        np.sin(2 * np.pi * 5 * timestamps)
+        + np.sin(2 * np.pi * 10 * timestamps)
+        + np.sin(2 * np.pi * 15 * timestamps)
+        + np.sin(2 * np.pi * 20 * timestamps)
+    )
+
+    data1 = np.array([signal, signal]).T
+    signal = Signal(data=data1, timestamps=timestamps, rate=100)
+    return signal
+
+
+def test_call_invalid_input_parameters():
+    with pytest.raises(NotImplementedError):
+        analyzer = SpectrumAnalysisBase(window_length_for_welch=4)
+        analyzer(mock_signal_generator())
+
+    with pytest.raises(ValueError):
+        SpectrumAnalysisWelch(window_length_for_welch=-1)
+
+    with pytest.raises(ValueError):
+        SpectrumAnalysisWelch(window_length_for_welch=0.75)
+
+
+def test_call_return_shape():
+    analyzer = SpectrumAnalysisWelch(window_length_for_welch=4)
+    result = analyzer(mock_signal_generator())
+    freqs, psd = result
+
+    assert isinstance(result, tuple)
+    assert len(result) == 2
+    assert isinstance(freqs, np.ndarray)
+    assert isinstance(psd, np.ndarray)
+    assert freqs.shape[0] == 801
+    assert psd.shape[1] == 2
+
+
+@pytest.mark.parametrize(
+    "analysis_class",
+    [SpectrumAnalysisWelch, SpectrumAnalysisPeriodogram],
+)
+def test_spectrum_analysis_computation(analysis_class) -> None:
+    analysis = analysis_class()
+    freqs, psd = analysis(mock_signal_generator())
+
+    for channel in range(2):
+        psd_channel = psd[:, channel]
+        total_psd_sum = np.sum(psd_channel)
+        expected_freqs = [5, 10, 15, 20]
+
+        range_psd_sum = 0
+        for exp_freq in expected_freqs:
+            range_psd_sum += np.sum(
+                psd[(freqs >= exp_freq - 1) & (freqs <= exp_freq + 1)]
+            )
+            idx = np.argmin(np.abs(freqs - exp_freq))
+            spike_psd = psd[idx, channel]
+            # Test the peaks of psd is as expected
+            assert np.isclose(freqs[idx], exp_freq)
+        # Test the psd around peaks comprises more than 90% of total psd
+        assert range_psd_sum / total_psd_sum >= 0.9
+
+        for i, freq in enumerate(freqs):
+            if all(abs(freq - exp_freq) > 1 for exp_freq in expected_freqs):
+                assert psd[i, channel] < 0.05 * spike_psd
+
+
+@pytest.mark.parametrize(
+    "analysis_class",
+    [SpectrumAnalysisWelch, SpectrumAnalysisPeriodogram],
+)
+def test_different_sampling_rates(analysis_class):
+    for rate in [50, 200, 500]:
```

Comment:
This test looks clean. I would use pytest.mark.parametrize for `rate` as well

Reply:
Sure, have used `pytest.mark.parametrize` for `rate`.